src/main/java/org/venice/beachfront/bfapi/controllers/OAuthController.java

```diff
@@ -63,26 +68,37 @@ public String oauthCallback(@RequestParam("code") String authCode, HttpSession s
 		String uiRedirectUri = UriComponentsBuilder.newInstance().scheme("https").host(this.domain).queryParam("logged_in", "true").build()
 				.toUri().toString();
 
-		session.setAttribute("api_key", userProfile.getApiKey());
+		response.addCookie(createCookie(userProfile.getApiKey(), COOKIE_EXPIRY_SECONDS));
 		response.setStatus(HttpStatus.FOUND.value());
 		response.setHeader("Location", uiRedirectUri);
 		return "Authentication successful. Redirecting back to application...";
 	}
 
 	@RequestMapping(path = "/oauth/logout", method = RequestMethod.GET, produces = { "text/plain" })
 	@ResponseBody
-	public String oauthLogout(HttpSession session, HttpServletResponse response) throws UserException {
-		// Remove cookie
-		session.invalidate();
+	public String oauthLogout(HttpServletResponse response, Authentication authentication) throws UserException {
+		// Server-side invalidation of API Key
+		userProfileService.invalidateKey(userProfileService.getProfileFromAuthentication(authentication));
 
 		// Construct redirect url for server side logout
 		final String uiUrl = "beachfront." + domain;
 		// Forward user to server side logout
-		String logoutRedirectUri = UriComponentsBuilder.fromUriString(oauthLogoutUrl).queryParam("end_url", uiUrl)
-				.build().toUri().toString();
+		String logoutRedirectUri = UriComponentsBuilder.fromUriString(oauthLogoutUrl).queryParam("end_url", uiUrl).build().toUri()
+				.toString();
 
+		// Clear the session cookie
+		response.addCookie(createCookie(null, 0));
 		response.setStatus(HttpStatus.FOUND.value());
 		response.setHeader("Location", logoutRedirectUri);
 		return "Logging out. Redirecting to oauth logout...";
 	}
+
+	private Cookie createCookie(String apiKey, int expiry) {
+		Cookie cookie = new Cookie(COOKIE_NAME, apiKey);
+		cookie.setDomain(String.format("*.%s", domain));
```

Comment:
https://serverfault.com/a/497774

According to this, the `*` is not necessary. This line should be:

    cookie.setDomain(String.format(".%s", domain));

Reply:
yeah it might not even need the leading dot